src/frontend/src/lib/components/convert/ConvertInputAmount.svelte

```diff
@@ -0,0 +1,84 @@
+<script lang="ts">
+	import { IconClose } from '@dfinity/gix-components';
+	import { debounce, isNullish, nonNullish } from '@dfinity/utils';
+	import { BigNumber } from '@ethersproject/bignumber';
+	import InputCurrency from '$lib/components/ui/InputCurrency.svelte';
+	import type { Token } from '$lib/types/token';
+	import { invalidAmount } from '$lib/utils/input.utils';
+	import { parseToken } from '$lib/utils/parse.utils';
+
+	export let token: Token;
+	export let amount: number | undefined = undefined;
+	export let disabled: boolean | undefined = undefined;
+	export let customValidate: (userAmount: BigNumber | undefined) => void = () => {};
+	export let error: Error | undefined = undefined;
+	export let placeholder = '0.00';
+
+	const onReset = () => {
+		amount = undefined;
+	};
+
+	const validate = () => {
+		const parsedValue = !invalidAmount(amount)
+			? parseToken({
+					value: `${amount}`,
+					unitName: token.decimals
+				})
+			: undefined;
+
+		customValidate(parsedValue);
+	};
+
+	const debounceValidate = debounce(validate, 300);
+
+	$: amount, token.decimals, debounceValidate();
+</script>
+
+<div class="convert-input-amount text-3xl font-bold" class:error={nonNullish(error)}>
+	<InputCurrency
```

Comment:
no placeholder?

Reply:
Placeholder is passed as a prop on the line 46.

Reply:
Ah cool.

Speaking of, `0.00` is a correct default place holder?! Two decimals per default, always?

Reply:
Ok. I was just asking myself if the number of decimals should equals the token decimals.